binary_to_decimal/binary_to_decimal.js

```diff
@@ -0,0 +1,53 @@
+// ----------------------------------
+// Function binaryToDecimal
+
+function binaryToDecimal(binary_number){
+    const length = binary_number.toString().length; 
+    const string = binary_number.toString();
+    let binaryVal = true; //default value
+
+    for(let i = 0; i < length; i++){  //binary check - compare string
+        if(string[i] == "0" || string[i] == "1"){}
+        
+        else {
+            binaryVal = false;            
+        }
+    }
+
+    if(binaryVal == false){
+        alert("Wrong input number, the program needs to be restarted.");
+    }
+
+// ----------------------------------
+// Conversion to decimal number
+
+    if(binaryVal){
+    let power = length - 1;
+    let finalOut = 0;
+    let tempNum = 0;
+
+
+    for(i = 0; i < length; i++){
+
+        tempNum = parseInt(string[i]); //convert string character to integer
+        finalOut = finalOut + tempNum*Math.pow(2, power);  // calculate number and add to total      
```

Comment:
Jestli jste mohli využít knihoven, jako například Math, tak v pořádku. Kdyby ses chtěl vyhnout knihovně, lze zapsat jako 
```javascript
finalOut = finalOut + tempNum * (2**power);
```
To lze ještě zkrátit
```javascript
finalOut += tempNum * (2**power);
```

Reply:
Áno, to je ten 'C-like' postup, ktorý si spomínal už pred tým, v C-čku nie je takýto double asterisk exponential, len _math.h_.

Reply:
Jop, v JS dříve tento zápis taky nebyl, převzali inspiraci z Python